### PR TITLE
Feat/target toggle component

### PR DESCRIPTION
**Pull request by nym1k**
Target Toggle Component on homepage

- Updated structure of component to be more semantic
- Updated styles as per designs from Iman
- SDGs now use name as a label (shorter text e.g., Goal 1) and description as a title (longer text e.g., End poverty in all its forms everywhere.)

### CODE REVIEW

**Review by WillKocur — CHANGES_REQUESTED** [review]
Looking good! Some small comments below:

1. The Label and title (homepage-targets__title) are not always present. When either one isn’t there, could you remove the element - I think it’s pushing off alignment although hard to see with no images!
2. Can you add shadow around selected sdg as in designs
3. Home targets card description needs some styling on MEA tab at top of card - looks like maybe it just needs popping in p tags actually.
4. The main tabs (SDGs, MEAs, Themes, National Indicators) have a small gap in the breakpoints where they wrap onto two lines. Might be worth running by Iman to see if she wants to reduce the font/padding size at all to prevent that. Difficult with only two breakpoints… 
Screenshot 2022-11-16 at 16 01 50
5. On safari the homepage-targets__logo goes really wide, however this may be just an issue when the image doesn’t load. Did this happen for you:
Screenshot 2022-11-16 at 16 04 11


I popped some optional comments in the code - only if you have time, and seems like things are quite rushed, so happy for them to be left if you don't.

File: app/assets/stylesheets/components/browse.scss [hunk 1]
```diff
@@ -1,42 +1,42 @@
...
+      font-size: 0.75rem;
+      padding: 0.5625rem 0.75rem;
     }
 
     &.is-selected {
-      font-weight: 700;
       background-color: $white;
-      border: solid 1px $border--search-bar;
-      border-bottom: solid 1px $white;
+      color: $prussian-blue;
```
[the remark this answers is not in the document]
Comment (nym1k) [reply app/assets/stylesheets/components/browse.scss:39]: Updated

File: app/assets/stylesheets/components/homepage-targets.scss [hunk 2]
```diff
@@ -1,103 +1,158 @@
...
 
-    @include m(flex) {
-      @include flexbox;
-      @include align-items(center);
+    @include only-mobile {
+      display: none;
+    }
+
+    @include m(round) {
+      border-radius: 100%;
```
[the remark this answers is not in the document]
Comment (nym1k) [reply app/assets/stylesheets/components/homepage-targets.scss:27]: Doesn't seem to be, i've deleted it

File: app/assets/stylesheets/components/targets-browse.scss [hunk 3]
```diff
@@ -1,103 +1,86 @@
...
     }
+  }
 
-    @include m(sdg) {
-      width: vr(8);
-      height: vr(8);
-      margin-bottom: vr(2);
-    }
+  @include e(goals-items) {
+    margin: 0 0 vr(-2) vr(-2);
```
[the remark this answers is not in the document]
Comment (nym1k) [reply app/assets/stylesheets/components/targets-browse.scss:33]: Updated to use a variable

File: app/assets/stylesheets/components/targets-browse.scss [hunk 4]
```diff
@@ -1,103 +1,86 @@
...
+    margin-left: vr(2);
+    width: 5.35rem;
+    height: 5.35rem;
+
+    position: relative;
 
     &.is-selected {
       &:after {
         content: "";
-        position: absolute;
```
[the remark this answers is not in the document]
Comment (nym1k) [reply app/assets/stylesheets/components/targets-browse.scss:85]: Couldn't see the properties in the design, but have them now so added it.

File: app/assets/stylesheets/components/themes-browse.scss [hunk 5]
```diff
@@ -1,56 +1,100 @@
...
   @include e(theme) {
-    cursor: pointer;
-    @include span-columns(4);
-    @include omega(3n);
-    margin-right: 0;
     @include flexbox;
     @include align-items(center);
+    cursor: pointer;
+    margin-right: 0;
     padding: vr(1) vr(1.6666666);
```
[the remark this answers is not in the document]
Comment (nym1k) [reply app/assets/stylesheets/components/themes-browse.scss:64]: This wasn't my code, but when I try that it seems to work so I will use it.

File: app/assets/stylesheets/components/themes-browse.scss [hunk 6]
```diff
@@ -1,56 +1,100 @@
...
+    background-color: $white;
     margin-right: vr(.75);
+    padding: vr(.75);
+    width: vr(5.75);
+    height: vr(5.75);
+
+    flex: 0 0 vr(5.75);
+
+    img {
+      object-fit: contain;
```
[the remark this answers is not in the document]
Comment (nym1k) [reply app/assets/stylesheets/components/themes-browse.scss:83]: Yes, we no longer support IE11

File: app/assets/stylesheets/variables/typography.scss [hunk 7]
```diff
@@ -157,26 +157,53 @@ $font-book: (
...
+    font-weight: 700,
+    line-height: 1.4,
+    color: $dark
   ),
+  goal-label: (
+    font-family: 'Open Sans',
+    font-size: 0.875rem,
+    font-weight: 600,
+    line-height: 1.3,
+    color: $black  ),
```
[the remark this answers is not in the document]
Comment (nym1k) [reply app/assets/stylesheets/variables/typography.scss:172]: Fixed

File: app/assets/stylesheets/components/homepage-targets.scss [hunk 8]
```diff
@@ -1,103 +1,158 @@
...
+    background-color: $white-two;
+    padding: vr(4);
+
+    @include only-mobile {
+      padding: vr(2);
+    }
   }
 
-  @include e(target-subheader) {
-    font-family: Merriweather;
```
[the remark this answers is not in the document]
Comment (nym1k) [reply app/assets/stylesheets/components/homepage-targets.scss]: I have pulled them in and tried to utilise them were possible, I have had to modify them slightly to make reusing possible.

**Comment by nym1k** [comment]
Hi @WillKocur, I have had to modify your mixins slightly to better adapt them for use in multiple places. I have also reworked how the link with arrow icon works and put this in to a button mixin.

**Comment by nym1k** [comment]
Also, from your list I am not seeing number 5 in safari, perhaps this is because I have the images loaded?

**Comment by WillKocur** [comment]
> Also, from your list I am not seeing number 5 in safari, perhaps this is because I have the images loaded?

Good good - I assume it's just when the image doesn't load then.

**Comment by WillKocur** [comment]
@nym1k Happy with these changes - I think the mixins look good 👍  Just the one thing that is a little off after reworking the styles:

The button on the resources page is a little different (on the left and not an arrow, also has a different class on the icon). So the mixin doesn't quite work for this one (see padding for vertical alignment and padding).

Perhaps just a case of having different classes/mixins for the two button/link types...

Screenshot 2022-11-17 at 18 42 49

**Comment by nym1k** [comment]
Reworked the search page results link element to allow for multiple types of link with icon

**Pull request merged**